### PR TITLE
Add reference to number control

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds a reference to the NumberControl component.

Screen Recording on 2024-07-11 at 08-27-40

Partially addresses https://github.com/woocommerce/woocommerce/issues/46768.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure `Try the new product editor (Beta)` is enabled at `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New.
3. Add a product name, and go to the Shipping tab.
4. Add a 0 as the `Height` value and go to the General tab.
5. Click Publish.
6. A snackbar with the message: `Product name is required. View error` should appear.
7. Press `View error`. You should be redirected to the Shipping tab with the `Height` field focused.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/hooks/use-error-handler.ts [hunk 1]
```diff
@@ -65,7 +65,9 @@ function getErrorPropsWithActions(
 					navigateTo( {
 						url: getUrl( errorContext ),
 					} );
-					focusByValidatorId( validatorId );
+					setTimeout( () => {
```
Comment (nathanss) [inline packages/js/product-editor/src/hooks/use-error-handler.ts]: Why was this added? Was it not working with the `InputControl` without the setTimeout?

**Review by nathanss — DISMISSED** [review]
Test instructions work as expected!

Left a question about the use of `setTimeout`.

File: packages/js/product-editor/src/hooks/use-error-handler.ts (hunk 1 above)
Comment (octaedro) [reply packages/js/product-editor/src/hooks/use-error-handler.ts]: @nathanss Yes, we need to focus on the component a few milliseconds later in order to have the component printed on the screen. Otherwise, it won't work.

File: packages/js/product-editor/src/components/number-control/number-control.tsx [hunk 2]
```diff
@@ -27,6 +29,7 @@ import { useNumberInputProps } from '../../hooks/use-number-input-props';
 import { Label } from '../label/label';
 
 export type NumberProps = {
+	identifier?: string;
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/number-control/number-control.tsx]: We should maintain component consistency so this should be renamed as `id`

File: packages/js/product-editor/src/components/number-control/number-control.tsx [hunk 3]
```diff
@@ -47,164 +50,175 @@ export type NumberProps = {
...
+					?.parentElement?.contains( event.relatedTarget )
+			) {
+				setIsFocused( false );
+				onBlur?.();
+			}
+		};
+		const inputProps = useNumberInputProps( {
+			value: value || '',
+			onChange,
+			onFocus: () => {
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/number-control/number-control.tsx]: Remember we are trying to remove the arrow functions when possible, and in this case is not necessary. So this can be changed to
```
onFocus() { ... }
```
instead of
```
onFocus: () => {}
```

File: packages/js/product-editor/src/components/number-control/number-control.tsx [hunk 4]
```diff
@@ -47,164 +50,175 @@ export type NumberProps = {
...
 
-	const isInitialClick = useRef< boolean >( false );
+		const isInitialClick = useRef< boolean >( false );
 
-	const incrementValue = () => {
-		const newValue = parseFloat( value || '0' ) + increment;
-		if ( newValue >= min && newValue <= max )
-			onChange( String( newValue ) );
-	};
+		const incrementValue = () => {
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/number-control/number-control.tsx]: this can be changed to a normal function too

File: packages/js/product-editor/src/components/number-control/number-control.tsx [hunk 5]
```diff
@@ -47,164 +50,175 @@ export type NumberProps = {
...
 
-	const handleIncrement = ( thisStep: number ) => {
-		const newValue = parseFloat( value || '0' ) + thisStep;
-		if ( newValue >= min && newValue <= max ) {
-			onChange( String( parseFloat( value || '0' ) + thisStep ) );
-			setIncrement( thisStep );
-			isInitialClick.current = true;
-		}
-	};
+		const resetIncrement = () => setIncrement( 0 );
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/number-control/number-control.tsx]: this can be changed to a normal function too

File: packages/js/product-editor/src/components/number-control/number-control.tsx [hunk 6]
```diff
@@ -47,164 +50,175 @@ export type NumberProps = {
...
-				isValidElement( label ) ? (
-					label
-				) : (
-					<Label
-						label={ label as string }
-						required={ required }
-						tooltip={ tooltip }
-					/>
-				)
+		const handleIncrement = ( thisStep: number ) => {
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/number-control/number-control.tsx]: this can be changed to a normal function too

**Review by mdperez86 — CHANGES_REQUESTED** [review]
I left one comment about the `id` that I think it's required for this PT to be merged.
Other suggestions are not required but I talked about avoiding the arrow functions when possible, all the arrow functions touched in this PR can be changed to normal functions without any problem.

File: packages/js/product-editor/src/components/number-control/number-control.tsx (hunk 3 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/number-control/number-control.tsx:94]: That line was affected by the linter, because I added a function `forwardRef` [on top](https://github.com/woocommerce/woocommerce/pull/49357/files#diff-36fe6f86d720a06f7d285658f2405bb26704e6f3cc81b621e874ca950f5cdc68R53). I like the idea of getting ride of not necessary arrow functions but I would tackle those changes in a follow-up PR.

File: packages/js/product-editor/src/components/number-control/number-control.tsx (hunk 2 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/number-control/number-control.tsx]: Done in [e8040b4](https://github.com/woocommerce/woocommerce/pull/49357/commits/e8040b4120ea69b1a328b70568167a1cd602db91)

File: packages/js/product-editor/src/components/number-control/number-control.tsx (hunk 3 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/number-control/number-control.tsx]: Done in the commit [f9f38dc](https://github.com/woocommerce/woocommerce/pull/49357/commits/f9f38dcf79b05c57e106924f07a3459e984b9e12)

File: packages/js/product-editor/src/components/number-control/number-control.tsx (hunk 5 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/number-control/number-control.tsx]: Done in [c564ba6](https://github.com/woocommerce/woocommerce/pull/49357/commits/c564ba63326cb5a8a58b510942f9704b1ddbdc0d)

File: packages/js/product-editor/src/components/number-control/number-control.tsx (hunk 6 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/number-control/number-control.tsx]: Done in [f9f38dc](https://github.com/woocommerce/woocommerce/pull/49357/commits/f9f38dcf79b05c57e106924f07a3459e984b9e12)

File: packages/js/product-editor/src/components/number-control/number-control.tsx (hunk 4 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/number-control/number-control.tsx]: Done in [f9f38dc](https://github.com/woocommerce/woocommerce/pull/49357/commits/f9f38dcf79b05c57e106924f07a3459e984b9e12)

**Review by mdperez86 — APPROVED** [review]
Tested this with cross tab validations an it does worked, the NumberControl is receiving the focus correctly for the Dimension fields.

Nice job @octaedro !!!
